### ISSUE TITLE
Enhance Window Popout Features

### ISSUE THREAD

**Issue by mathuo** [mathuo/dockview#469]
- Reposition panels in the same position after the popout window is closed when possible
- Expose the underlying Window/Document object for Popout windows
- General internal improvements to the handling of window popouts
- Support 'always' rendering mode in Popouts

**Comment by sachnk** [comment]
Hey @mathuo --

Any chance this would be included in the 1.9.3 release of dockview? And if so, any rough ideas on when that might be?

Trying to figure out on my end if I should build against my fork or wait for the official release. Appreciate all the great work you're doing on this library!

Thanks!

**Comment by mathuo** [comment]
I can put this into `1.9.3` within the next couple of days. 

Could you describe your use-case for the lifecycle events  just in-case I can think of any extra additons that may be useful for these popout windows too?

**Comment by sachnk** [comment]
Awesome, great to hear!

The use-case is specific to issues with popouts [described in FlexLayout](https://github.com/caplin/FlexLayout?tab=readme-ov-file#floating-tabs-popouts). When a new window is opened, and a react portal is mapped to the new window's DOM, libraries that rely on referencing `window.document` tend to break.

We specifically use [react-laag](https://www.react-laag.com/) for drop-down menus and such. We pass the current `window.document` to `react-laag` via its props. The FlexLayout documentation suggests using `this.selfRef.current.ownerDocument.defaultView` as way to get the current `window.document`. However, in practice I've found that this doesn't always work. In particular, once a react component is mapped to the new window, I've found this value to be incorrect at times (though sometimes it's right).

That value can't ever be wrong, because it means a drop-down menu would render off-window, which is pretty bad. To eliminate that possibility, I use the `onWindow*` events to explicitly set the `window.document` when passed to libraries like react-laag.

**Comment by mathuo** [comment]
Ok nice - that makes sense.

I'm going to also provide a way to obtain the "location" of the panel in this PR (i.e. is it within the main grid, a floating group or a popouted group).

For example you could have a React hook that tracked the correct `window` object:

```tsx
function usePopoutWindowContext(api: DockviewPanelApi): Window {
    const [isPopout, setIsPopout] = React.useState<boolean>(
        api.location === 'popout'
    );

    React.useEffect(() => {
        const disposable = api.onDidLocationChange((event) => {
            setIsPopout(event.location === 'popout');
        });

        return () => {
            disposable.dispose();
        };
    });

    const windowContext = React.useMemo(() => window, [isPopout]);

    return windowContext;
}
```

**Comment by mathuo** [comment]
Released in 1.10.0 https://dockview.dev/
